src/App.tsx

```diff
@@ -1,14 +1,47 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { Options } from './types/Options';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [selectOption, setSelectOption] = useState<Options>('all');
+  const [selectPost, setSelectPost] = useState<Todo | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .finally(() => setLoading(false));
+  }, []);
+
+  const filterTodos = () => {
+    const todoFilter = todos.filter(todo =>
+      todo.title.toLowerCase().includes(query.toLowerCase()),
+    );
+
+    if (selectOption === 'active') {
+      return todoFilter.filter(todo => !todo.completed);
+    }
+
+    if (selectOption === 'completed') {
+      return todoFilter.filter(todo => todo.completed);
+    }
```

Comment:
Use a switch case instead of if conditions. Also create a enum for 'all', 'active', 'completed' and use it everywhere